driver-core/src/test/java/com/datastax/driver/core/IndexMetadataTest.java

```diff
@@ -73,167 +95,187 @@
     }
 
     @Test(groups = "short")
-    public void should_not_flag_text_column_index_type() {
+    public void should_create_metadata_for_simple_index() {
         String createValuesIndex = String.format("CREATE INDEX text_column_index ON %s.indexing (text_column);", keyspace);
         session.execute(createValuesIndex);
         ColumnMetadata column = getColumn("text_column");
         IndexMetadata index = getIndex("text_column_index");
-        assertThat(index).hasName("text_column_index")
-            .isNotKeys()
-            .isNotFull()
-            .isNotEntries()
+        assertThat(index)
+            .hasName("text_column_index")
+            .hasTable(column.getTable())
             .isNotCustomIndex()
-            .hasSingleColumn(getColumn("text_column"))
+            .hasTarget("text_column")
             .asCqlQuery(createValuesIndex);
-        assertThat(index).isSameAs(column.getIndex("text_column_index"));
-        assertThat(column).hasSingleIndex(index);
-        assertThat(column).isSameAs(index.getColumn("text_column"));
+        assertThat(column.getTable()).hasIndex(index);
+    }
+
+    @Test(groups = "short")
+    public void should_create_metadata_for_simple_index_on_mixed_case_column() {
```

Comment:
This one is failing on <= 2.1 for some reason, though looks like it should work...looking into it.